### ISSUE TITLE
Safari support

### ISSUE THREAD

**Issue by jrk** [roam-unofficial/roam-toolkit#44]
I know this is a nontrivial request, as it has a very different extension model, but I would love to see support for Safari. It is _so_ much better on macOS than Chrome.

In support of this, and other potential integrations (e.g., the unofficial iOS app, as userscripts in site-specific browsers or Electron packages, as bookmarklets for use anywhere, etc.), it would be great to see much of this functionality organized into simple library modules, somewhat decoupled from the browser extension infrastructure.

**Comment by Stvad** [comment]
Yeah it's not trivial. I don't use Safari, but I've heard that for newest version you can only develop extensions in Swift. So its totally incompatible with other extension platforms

**Comment by jrk** [comment]
Extensions are delivered via apps and the mac/iOS "extension" mechanism,
which is exposed via Objective-C and Swift APIs, but that's just the glue
for loading things. The meat of the work is generally all Javascript and
CSS injection, applying actions by running Javascript, etc.

In short, it's the scaffolding and how you add buttons to the toolbar, etc.
that's different, not most of the core logic for interacting with and
extending the web content itself.

**Comment by imagine** [comment]
Apple just released a new extensions API and a Chrome extension converter. It’ll work on Catalina once Safari 12 comes out. I’ll explore this for y’all.

**Comment by Stvad** [comment]
thank you @imagine!
